pilot/pkg/bootstrap/server.go

```diff
@@ -211,7 +211,8 @@ func NewServer(args *PilotArgs) (*Server, error) {
 	}
 	// Initialize workload Trust Bundle before XDS Server
 	s.workloadTrustBundle = tb.NewTrustBundle()
-	s.XDSServer = xds.NewDiscoveryServer(e, args.Plugins, args.PodName)
+	s.XDSServer = xds.NewDiscoveryServer(e, args.Plugins, args.PodName,
```

Comment:
It's not a good practice to keep adding params to New -first because it's not backward compatible ( and the code may be used in other places), second readability. 

Instead, add workloadTrustBundle to Environment ( it seems something that should be there anyways ). 
Plugins and PodName should have also been added there, but too late.

Reply:
make sense. will add to env